hypershift-operator/metrics.go

```diff
@@ -68,6 +94,19 @@ func newMetrics(client crclient.Client, log logr.Logger) *hypershiftMetrics {
 			Name: "hypershift_nodepools_size",
 			Help: "Number of replicas associated with a given NodePool",
 		}, []string{"name", "platform"}),
+		// hypershiftOperatorInfo is a metric to capture the current operator details of the management cluster
+		hypershiftOperatorInfo: prometheus.NewGaugeFunc(prometheus.GaugeOpts{
```

Comment:
you're declaring the variable but never writing it, see `NewGaugeFunc` docs. There fore the test is failing to find the metric because is never written. 

```
// NewGaugeFunc creates a new GaugeFunc based on the provided GaugeOpts. The
// value reported is determined by calling the given function from within the
// Write method. Take into account that metric collection may happen
// concurrently. Therefore, it must be safe to call the provided function
// concurrently.
//
// NewGaugeFunc is a good way to create an “info” style metric with a constant
// value of 1. Example:
// https://github.com/prometheus/common/blob/8558a5b7db3c84fa38b4766966059a7bd5bfa2ee/version/info.go#L36-L56
func NewGaugeFunc(opts GaugeOpts, function func() float64) GaugeFunc {
	return newValueFunc(NewDesc(
		BuildFQName(opts.Namespace, opts.Subsystem, opts.Name),
		opts.Help,
		nil,
		opts.ConstLabels,
	), GaugeValue, function)
}
```

Reply:
issue is the query in the e2e is filtering by hc name so would never satisfy this metric, you need to have a query for this metric

Reply:
Yep, thank you @enxebre @muraee for the findings. I'm filtering now for the container name, this way the metric will not be modified at all, if that does not work, I will add a new static label to identify uniquely.